### DISCUSSION TITLE
[ZK Stack] - <Can anyone tell me what the bootloader does, and why zksync lite doesn't have one?>

### DISCUSSION THREAD

**Discussion by zk616153275** [zkSync-Community-Hub/zksync-developers#380] [ZK Stack] [answered]
### Module Affected

- [X] System contracts (era-system-contracts)
- [X] Smart contracts (era-contracts)
- [X] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.75.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

I was reading zksync-era code in the process, encountered bootloader, I read the documentation of the zk stack, still do not understand its role, it feels like it is used for auxiliary circuits of some witness processing, but it has a corresponding contract address, I am a little difficult to understand

### Expected Behavior

.

### Current Behavior

.

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by dutterbutter** [comment, marked as answer]
@zk616153275 thanks for your interest! 

The bootloader in the context of zkSync is our way to handle a lot of transactions all at once, instead of doing them one by one like on Ethereum. On Ethereum, normally, each transaction is picked, checked, charged a fee, executed, and then its results are added to the system. This happens one after the other until there's no more room in the block.

But on zkSync, doing it this way would be way too slow because each transaction needs a special check called a "proving workflow". So, the bootloader comes in to make things faster. It takes a whole bunch of transactions together, along with some extra info, and treats them as if they were just one big transaction. This way, instead of doing all the steps for each transaction, it does it all in one go. It's kind of like how in EIP4337 there's the EntryPoint that also groups transactions for processing. This makes everything much more efficient on zkSync.

For more information regarding the bootloader check out this doc: 👉  https://docs.zksync.io/zk-stack/components/zkEVM/bootloader.html

As it relates to zkSync lite, zkSync Lite is a stripped-back version of what we now know as zkSync Era, primarily designed to facilitate basic payment solutions through its zk-rollup architecture. zkSync era is much more a complete solution that includes contract development, native AA and other features.

**Comment by npam90** [comment]
The zkSync bootloader is a foundational component responsible for initializing zkSync contracts and circuits, managing system configuration, and facilitating various operational aspects of the zkSync protocol. It plays a crucial role in the bootstrapping and operation of the zkSync system.

**Comment by deosovitis** [comment]
The user is seeking clarification on the role of the bootloader in the zksync-era codebase, as they are having difficulty understanding its purpose despite reading the documentation. They note that it has a corresponding contract address and suspect it may be used for witness processing. No additional details or repository link were provided, and the user states that they conducted prior research on similar issues without success.
